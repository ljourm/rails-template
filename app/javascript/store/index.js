import Vue from "vue"
import Vuex from "vuex"
Vue.use(Vuex)

import base from "./modules/base"
import api from "./modules/api"
import user from "./modules/user"
import users from "./modules/users"
import roles from "./modules/roles"

export default new Vuex.Store({
  strict: process.env.NODE_ENV === "development",
  modules: {
    base: base,
    api: api,
    user: user,
    users: users,
    roles: roles,
  },
})
