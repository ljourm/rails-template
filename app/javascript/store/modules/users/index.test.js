import store from "./index"

const mutations = store.mutations
const actions = store.actions

describe("store", () => {
  test("snapshot", () => {
    expect(store).toMatchSnapshot()
  })
})

describe("mutations", () => {
  describe("SET_USERS", () => {
    it("normal", () => {
      const post = [{ name: "test" }]
      const state = {
        users: {},
      }

      mutations.SET_USERS(state, post)
      expect(state).toEqual({ users: post })
    })
  })
})

describe("actions", () => {
  describe("fetch", () => {
    it("APIのレスポンスにユーザが一つある時", async () => {
      const users = [{ name: "test" }]
      const data = {
        users: users,
      }

      const commit = jest.fn()
      const dispatch = jest.fn().mockReturnValueOnce(data)

      await actions.fetch({ commit, dispatch })

      expect(commit).toHaveBeenCalledWith("SET_USERS", users)
      expect(dispatch).toHaveBeenCalledWith(
        "api/fetch",
        { url: "/api/v1/users" },
        { root: true }
      )
    })

    it("APIのレスポンスが不正な時", async () => {
      const data = null

      const commit = jest.fn()
      const dispatch = jest.fn().mockReturnValueOnce(data)

      await actions.fetch({ commit, dispatch })

      expect(commit).toHaveBeenCalledWith("SET_USERS", [])
    })
  })
})
