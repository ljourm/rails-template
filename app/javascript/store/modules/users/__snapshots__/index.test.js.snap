// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store snapshot 1`] = `
Object {
  "actions": Object {
    "fetch": [Function],
  },
  "mutations": Object {
    "SET_USERS": [Function],
  },
  "namespaced": true,
  "state": Object {
    "users": Array [],
  },
}
`;
